File: vdb_gorm/models/BaseModel.go
package models

import "time"

// BaseModel GORM
type BaseModel struct {
	// GORM tự động quản lý CreatedAt và UpdatedAt nếu có tên này và kiểu time.Time
	// Thẻ `column` để đảm bảo tên cột là snake_case trong DB.
	// Thẻ `index` để tạo index.
	CreatedAt *time.Time `gorm:"column:created_at;index"`
	UpdatedAt *time.Time `gorm:"column:updated_at;index"`

	Description *string `gorm:"type:varchar(255)"` // vdb: db:"size:255"
}

File: vdotnet/VDotnetOrmTest/Models/BaseModel.cs
using System;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
public class BaseModel
{

    [Column(TypeName = "datetime2(3)")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    [Column(TypeName = "datetime2(3)")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public string? Description { get; set; }
}